kitsune/questions/tasks.py

```diff
@@ -30,30 +30,25 @@ def update_question_votes(question_id):
 
 
 @shared_task(rate_limit="4/s")
-def update_question_vote_chunk(data):
-    """Update num_votes_past_week for a number of questions."""
-
-    # First we recalculate num_votes_past_week in the db.
-    log.info("Calculating past week votes for %s questions." % len(data))
-
-    ids = ",".join(map(str, data))
-    sql = (
-        """
-        UPDATE questions_question q
-        SET num_votes_past_week = (
-            SELECT COUNT(created)
-            FROM questions_questionvote qv
-            WHERE qv.question_id = q.id
-            AND qv.created >= DATE(SUBDATE(NOW(), 7))
+def update_question_vote_chunk(question_ids):
+    """Given a list of questions, update the "num_votes_past_week" attribute of each one."""
+    from kitsune.questions.models import Question, QuestionVote
+
+    log.info("Calculating past week votes for %s questions." % len(question_ids))
+
+    past_week = (datetime.now() - timedelta(days=7)).replace(
+        hour=0, minute=0, second=0, microsecond=0
+    )
+
+    Question.objects.filter(id__in=question_ids).update(
+        num_votes_past_week=Subquery(
+            QuestionVote.objects.filter(question_id=OuterRef("id"), created__gte=past_week)
+            .order_by()
```

Comment:
do we need the `order_by`?

Reply:
I think so, at least to be safe, but I'm not absolutely certain. Here's my reference for using it: https://docs.djangoproject.com/en/4.2/topics/db/aggregation/#interaction-with-order-by. My thinking was that I need to clear any default ordering on the `QuestionVote`, which could mess-up the annotations (for example, the grouping by `question_id`). `QuestionVote` doesn't have a default ordering, but if one was added in the future, it could cause problems in a way that would be difficult to understand.